src/components/EnsureAdminAccess.tsx

```diff
@@ -0,0 +1,14 @@
+import { FC } from "react";
+import { useUserHasRole } from "../api/hooks";
+
+const EnsureAdminAccess: FC<{}> = ({ children }) => {
+  const isAdmin = useUserHasRole("admin");
+
+  if (!isAdmin) {
+    return <p>You do not have sufficient access to view this page.</p>;
+  } else {
+    return <>{children}</>;
+  }
+};
```

Comment:
In my case, when I go to `/config`, I see first: waiting for authentication --> waiting for configuration --> You do not have sufficient access to view this page --> config page with full access. So it's a bit confusing to me as I just saw the text: "You do not have sufficient access to view this page" and then I have full access to the page again. From the code, I understand that if I have admin right, I do not see the warning text above and if I see the warning text above, I will not see the config page. So I don't really understand what happened there.

Reply:
It's because the useUserHasRole hook always returns true or false, even when the request is still busy fetching. So until it comes back with a response saying you're admin, the default message is shown.

Personally I thought it was no big deal that the message is visible briefly.